hoomd/mpcd/Integrator.cc

```diff
@@ -97,6 +98,14 @@ void mpcd::Integrator::update(unsigned int timestep)
     // call the MPCD collision rule before the first MD step so that any embedded velocities are updated first
     if (m_collide)
         m_collide->collide(timestep);
+    // remove any virtual particles in the system
+    if (checkcollide)
+        {
+        if (m_mpcd_sys->getParticleData()->getNVirtual() > 0)
+            {
+            m_mpcd_sys->getParticleData()->removeVirtualParticles();
+            }
```

Comment:
I think it is safe to just call this even if there aren't virtual particles.
```suggestion
        m_mpcd_sys->getParticleData()->removeVirtualParticles();
```